hack/assets/providers.go

```diff
@@ -293,8 +293,20 @@ func importProviders() error {
 			return err
 		}
 
+		objs := []unstructured.Unstructured{}
+
+		// Explicitly remove all defaulting webhooks for the core provider.
+		// We don't need CAPI to set any default to the cluster object because
+		// we have a custom controller for reconciling it.
+		// For more information: https://issues.redhat.com/browse/OCPCLOUD-1506
+		for _, obj := range p.components.Objs() {
```

Comment:
These better be handled here https://github.com/openshift/cluster-capi-operator/blob/main/hack/assets/customizations.go#L51. The code also removes webhooks for machine, machinesets, etc., but it should only remove webhook from the generic clusters and all infra clusters.